Added Cypress and simulated few simple tests

k1eu opened the pull request.

k1eu committed:
test: added basic cypress tests

Wozniaxos commented on cypress/integration/countriesPaginationFirstDisabled_spec.js:
```diff
@@ -0,0 +1,13 @@
+        cy.contains('button','Dalej').click()
+      }
+      for (let i=0;i <2;i++) {
+        cy.contains('button','Wróć').click()
```
better to have custom data-testid attribute and get it by cy.get, in that scenario when text on the button will change, for example from Wróć to back, your tests will fail, but the logic will stil be working quite well

Wozniaxos commented on cypress/integration/countriesPaginationFirstDisabled_spec.js:
```diff
@@ -0,0 +1,13 @@
+      for (let i=0;i <2;i++) {
+        cy.contains('button','Dalej').click()
+      }
+      for (let i=0;i <2;i++) {
```
I would avoid any logic in tests, tests needs to bu stupid simple, don;t worry about too much code, tests are rather books than code :)

Wozniaxos commented on cypress/integration/countriesPaginationFirstDisabled_spec.js:
```diff
@@ -0,0 +1,13 @@
+describe('Country Click test', () => {
+    it('successfully clicks & loads first, third and last country', () => {
```
Where are you test in this test "loads first, third and last country"? :)

Wozniaxos commented on cypress/integration/countriesPaginationFirstDisabled_spec.js:
```diff
@@ -0,0 +1,13 @@
+describe('Country Click test', () => {
```
I would put all of this test in one describe, just describe "country list" or similiar

Wozniaxos commented on cypress/integration/countriesPaginationLastBlock_spec.js:
```diff
@@ -0,0 +1,9 @@
+describe('Country Click test', () => {
+    it('successfully clicks & loads first, third and last country', () => {
```
Same as above, i see only button testing here, not checking for countries showed

Wozniaxos commented on cypress/integration/countriesPaginationLastBlock_spec.js:
```diff
@@ -0,0 +1,9 @@
+describe('Country Click test', () => {
+    it('successfully clicks & loads first, third and last country', () => {
+      cy.visit('/countries')
+      for (let i=0;i <21;i++) {
```
Similiar as above

Wozniaxos commented on cypress/integration/countriesPaginationNext_spec.js:
```diff
@@ -0,0 +1,9 @@
+      cy.contains('button','Dalej').click()
+      cy.url().should('include','?page')
+      cy.get('h2')
+      cy.get('h5') // test error
```
data-testid :)

Wozniaxos commented on cypress/integration/countryClick_spec.js:
```diff
@@ -0,0 +1,13 @@
+describe('Country Click test', () => {
+    it('successfully clicks & loads first, third and last country', () => {
```
oh, maybe this test text was just copied :D test title is super important, i would say most important because this tells what you are testing inside

Wozniaxos commented on cypress/integration/homeToCountries_spec.js:
```diff
@@ -0,0 +1,7 @@
+describe('The Home Page To List', () => {
+    it('successfully loads & links to countries list', () => {
```
Title should be more like, "redirect to countries" because you don't check whether on countries page there is everything rendered properly, you just checked if url is ok

Wozniaxos commented on cypress/integration/homeToCountries_spec.js:
```diff
@@ -0,0 +1,7 @@
+describe('The Home Page To List', () => {
+    it('successfully loads & links to countries list', () => {
+      cy.visit('/') // change URL to match your dev URL
+      cy.contains('a','Go to Countries').click()
```
data test ids :)

Wozniaxos reviewed (changes requested):
Nice overall, i left some improvements comments

k1eu committed:
fix: added data-testid to elements that are being tested

simplified file structure of tests - merged countriesList testi into one file
got rid of for loops in tests

Wozniaxos commented on cypress/integration/countriesList_spec.js:
```diff
@@ -0,0 +1,47 @@
+  it("successfully clicks & loads first, third and last country", () => {
+    cy.visit("/countries");
+    cy.get("[data-testid=countryName]").first().click();
+    cy.get("[data-testid=countryName]");
```
dead get?

Wozniaxos commented on cypress/integration/countriesList_spec.js:
```diff
@@ -0,0 +1,47 @@
+    cy.get("[data-testid=countryName]");
+    cy.go("back");
+    cy.get("[data-testid=countryName]").eq(3).click();
+    cy.get("[data-testid=countryName]");
```
Same, how do you know that this is id from single country and not from counties list :) We usually name it like data-testid='country-list-item' etc etc

Wozniaxos commented on cypress/integration/countriesList_spec.js:
```diff
@@ -0,0 +1,47 @@
+    cy.get("[data-testid=countryName]");
+    cy.go("back");
+    cy.get("[data-testid=countryName]").last().click();
+    cy.get("[data-testid=countryName]");
```
same

Wozniaxos commented on src/pages/countries.js:
```diff
@@ -56,10 +56,10 @@ const Countries = (props) => {
       <p>Cinema page</p>
       <div>
-        <button onClick={handleOnClickPrevious} disabled={onFirstPage}>
+        <button data-testid="backButton" onClick={handleOnClickPrevious} disabled={onFirstPage}>
```
Let's write raw html attributes as `back-button` manner

k1eu committed:
fix: changed naming convention to kebab-case & separated list country-name ids from those inside country view
